### ISSUE TITLE
Licensing Winapp2

### ISSUE THREAD

**Issue by ROCKNROLLKID** [MoscaDotTo/Winapp2#86]
Robert and I have been discussing about licensing Winapp2. Robert said he wanted people to be able to use Winapp2 freely, but submit anything they do back to the community here. We would need a free license, and because winapp2 is just a text file, we would have to use a free documentation license. The best license for this situation I can think of is the CC-BY SA license, where if anyone modified Winapp2, they would have to license it under the same license with the same permissions. https://creativecommons.org/licenses/by-sa/3.0/

If everyone agrees, I will go ahead and license Winapp2.

**Comment by TZocker** [comment]
Its ok to do this.

But whats is with a later License change. 

whats the diffrent to that licences  (same only in higher Version) automatically update?
https://creativecommons.org/licenses/by-sa/4.0/

**Comment by the-super-coder** [comment]
Good idea. I think it means, if anyone wants to take benefit of WinApp2 and wants to modify something and publishes with their own name, then it should still stay free and open source. Right?

**Comment by ROCKNROLLKID** [comment]
Yes, it will have to be free and open-source.

**Comment by ShaneGowland** [comment]
Hello.

Some clear usage guidelines around the use of this file is a good thing, but there's a few things I'd like to raise for consideration.

Firstly, Creative Commons licenses imply ownership. Who holds this ownership? The file originated with an assortment of posters on the Piriform community forum who compiled several entries about a decade ago. I combined these into a single file and started offering it in 2008. Dozens of additional Piriform Forum and SingularLabs/TheWebAtom forum users also contributed entries; Robbie then maintained the file for a long time presumably receiving hundreds of entries from many other people. 

Which of these people retain ownership of the content they added? Does the chosen license allow you to retroactively override the license terms set by other contributors? For example, if someone said "here's an entry, do whatever the fuck you want with it," does your licensing it under anything other than the WTFPL license violate their intellectual property rights? 

Secondly, have any entries added by BleachBit ever been incorporated into the file? BleachBit is licensed under the GNU General Public License, which is considered to be a viral license; any project incorporating GPL code inherits the license. CC-BY-SA is also a viral license, which could lead to problems as you can't re-license something already covered by two competing viral licenses.

Finally, can you enforce a license on work that's entirely written in a proprietary configuration structure that's owned by Piriform? 

Does any of this really matter? Probably not. The chances of this ever being brought before a court is effectively nil, but it does pose one final question for consideration: is it worth the effort of licensing winapp2.ini if that license is going to be legally unenforceable?

I say "sure, whatever, go for it." If it makes it easier for others to know what they can and can't do with the file, there's probably no harm.

**Comment by ROCKNROLLKID** [comment]
Robert would most likely be announced ownership if it has to come to it. Basically, he wanted a license where if anyone did any modifications to it, he wanted them to give back to the main Winnap2 respo, while still having the freedom to use and the file and edit it all they way. I recommended the CC BY SA license because it seemed the best option for what he wanted. The WTFPL license allows anyone to do anything to it and that's not really what Robert wanted. Also, WTFPL is more for source code. We would need a documentation license instead.

The removed entries was moved back to the respo here and is managed here. I designed it now so that your project, as well as Avira and future projects can also use it, without worrying about losing entries due to them being added to CCleaner. So yes, we manage it now.

I don't know about other licenses, but from what I read from the CC BY SA license it has no problems with incorporating it in proprietary software, like yours or CCleaner. I am still waiting to hear back from Piriform though, to see how they feel about this change. I would like to get this done as soon as possible, as there are other projects looking into using Winapp2.

Since Winapp2 moved to GitHub, it has become more of a independent project. I don't want to say it is a full independent project, as I haven't confirmed such a thing, yet.

There is the matter of what you want to do with your tools you had published here, such as the CCDump and CCDebug?

**Comment by ShaneGowland** [comment]
Do whatever you think is best—I'm unconcerned either way. Just recognize that the collaborative way in which the file was created makes it virtually impossible to protect/license in a legally enforceable way.

As for the CCDump and CCDebug tools; you're free to do whatever you want with them what you want. I've left them purposefully [unlicensed](http://unlicense.org/).

**Comment by CSGalloway** [comment]
I think some of the list members would like to see links for WTFPL and CC
BY SA please.... any any other similar products please.

http://www.facebook.com/CSGalloway

**Comment by ROCKNROLLKID** [comment]
@az0 @ShaneGowland We had a meeting with Piriform today. We are waiting on the official word, but it is most likely this will be licensed under CC BY SA 4.0 license. Please consult the license for any changes that may affect your software.

https://creativecommons.org/licenses/by-sa/4.0/

**Comment by ROCKNROLLKID** [comment]
Just a update reminder that a little over a month ago I got in contact with Stephen Piriform about licensing Winapp2 and had sent him a email and he said he would contact the rest of the team and get back to us. I just threw him a PM message today to get a update on how progress is going. We are trying our best to get Winapp2 licensed.

@MoscaDotTo You are good to go with licensing the tools, though, as these are not made or associated with Piriform at all. If you want to use the same logic with the tools as you do with Winapp2, then the GNU GPLv3 license is probably the best way to go.
